lib/service/servicecfg/jamf_test.go

```diff
@@ -0,0 +1,103 @@
+/*
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package servicecfg
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestValidateJamfCredentials(t *testing.T) {
+	tests := []struct {
+		name         string
+		creds        *JamfCredentials
+		errAssertion require.ErrorAssertionFunc
```

Comment:
Suggestion:

```suggestion
		wantErr string // empty means no error wanted
```

and push the assert.ErrorContains to the test body? It should make for an easier-to-read table.